### PR TITLE
ESQL: Fix Max doubles bug with negatives and add tests for Max and Min

### PR DESCRIPTION
**Pull request by ivancea**
`MAX()` currently doesn't work with doubles smaller than `Double.MIN_VALUE` (Note that `Double.MIN_VALUE` returns the smallest non-zero positive, not the smallest double).

This PR adds tests for Max and Min, and fixes the bug (Detected by the tests).

Also, as the tests now generate the docs, replaced the old docs with the generated ones, and updated the Max&Min examples.

### CODE REVIEW

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractAggregationTestCase.java [hunk 1]
```diff
@@ -166,7 +168,9 @@ private void aggregateWithIntermediates(Expression expression) {
 
             Page inputPage = rows(testCase.getMultiRowFields());
             try {
-                aggregator.processPage(inputPage);
+                if (inputPage.getPositionCount() > 0) {
```
Comment (nik9000) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractAggregationTestCase.java:171]: This is fine for now, but I wonder if we should make `rows` make a `List<Page>` and we dump them all in. So if it spits out no pages that's cool. At some point we're going to have to signal "empty" with no results, but that *could* live in `rows`.

**Review by nik9000 — APPROVED** [review]

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractAggregationTestCase.java (hunk 1 above)
Comment (ivancea) [reply x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractAggregationTestCase.java]: Changed it already. Maybe check it; I'm creating pages of 1-10 elements. Went with the small values as some hand-made tests may have just a few elements. But it should still work either way, so whatever

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java [hunk 2]
```diff
@@ -230,27 +230,38 @@ protected final Page rows(List<TestCaseSupplier.TypedData> multirowFields) {
...
-        for (int i = 0; i < multirowFields.size(); i++) {
-            var field = multirowFields.get(i);
-            try (
-                var wrapper = BlockUtils.wrapperFor(
-                    TestBlockFactory.getNonBreakingInstance(),
-                    PlannerUtils.toElementType(field.type()),
-                    rowsCount
-                )
-            ) {
+        int pageSize = randomIntBetween(1, 10);
```
Comment (nik9000) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java:235]: 10,000 is a pretty normal size in production. Maybe `randomBoolean() ? between(1, 100) : Math.toIntExact(BlockFactory.DEFAULT_MAX_BLOCK_PRIMITIVE_ARRAY_SIZE.getBytes())`. Half the time we'll have degenerately small data and half the time it'll probably just be one page.

**Review by nik9000 — APPROVED** [review]

**Comment by nik9000** [comment]
backport incoming

**Pull request merged**